### ISSUE TITLE
Update phpunit for all repositories to at least 7.0

### ISSUE THREAD

**Issue by Hackwar** [joomla-projects/ATT#10]
What the title says.

**Comment by mbabker** [comment]
This must be asterisked, PHPUnit 7.0 will not run on PHP 7.1 or older, and a PHPUnit build must be able to run for PHP 5.3 and newer for all applicable resources.

The CMS needs to fix itself to be able to run PHPUnit across multiple major versions of that platform.  The Framework mostly does this, though I never added 7.0 to the PHP 5.3 compatible branches and the PHP 7.2+ branches all require PHPUnit 8.0 (minus the Database package due to needing to abandon the DbUnit extension).

**Comment by Hackwar** [comment]
Yes, the better title would probably have been "Update the freaking tests for all the neglected repos that we have". George complained about weblinks for example, which hasn't gotten an update in years in that regard.

**Comment by mbabker** [comment]
Weblinks is only neglected in its test suite?  Well, that's good news 🤣

**Comment by wilsonge** [comment]
Weblinks is now fine. I've been testing it the last few weeks - the tests needed some love but functionally the component itself worked with 3.9.11. And the testing also give or take now works after my commits two weeks ago. I just need to roll back the selenium version from 3.141 which has issues with the way they've moving to the w3c platform (but they work just fine on my macbook on firefox). So unless you've tested weblinks and have any better ideas you can stop trolling about weblinks :)

Your comments about phpunit and php 5.3 are totally correct tho so @Hackwar you absolutely need to take in account phpunit 7.0 is only for stuff that's shipping with j4
